### ISSUE TITLE
BOS Antminer S19 XP Miner is not returning data instead throwing error .

### ISSUE THREAD

**Issue by vijendtra** [UpstreamData/pyasic#119]
**Describe the bug**

**To Reproduce**
 Trying to get data from BraiinsOS firmware Antminer S19XP
 using this code `miner_data = await miner.get_data()`
 it return 
`Failed to fetch data for miner 10.XX.XX.XXX: LUCI web command failed: command=bos/info`

**Expected behavior**
It should return data without having any error. 

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Macbook
 - Version: Sanoma 14.4.1

**Miner Information (If applicable):**
 - Manufacturer: Antminer
 - Type: S19 XP
 - Firmware Type:  BraiinsOS
 - Firmware Version: Version: 2023-12-15-0-f43dca21-23.12-plus

**Additional context**

**Comment by UpstreamData** [comment]
Can you confirm the result of get_miner for that IP?  Want to make sure it's actually being identified.  In theory that miner shouldn't send a LUCI command, so it's likely failing the identification.

**Comment by vijendtra** [comment]
This is what i am getting `Unknown (BOS+): 10.XX.XX.171`

**Comment by UpstreamData** [comment]
Ok yep, failing to identify.  Let me make sure it's supported.

**Comment by vijendtra** [comment]
Thanks .

**Comment by vijendtra** [comment]
it's look like it has  really old version of Braiins OS+,I suspect they don't have the new GRPC based API @UpstreamData

**Comment by UpstreamData** [comment]
XP should be supported.  Can you send the output of `await pyasic.miner_factory.get_miner_model_braiins_os(IP)`?  It seems like the XP string might be a bit strange in this case, like something has been changed with it...

**Comment by vijendtra** [comment]
This is what i am getting `Unknown (BOS+)` @UpstreamData

**Comment by UpstreamData** [comment]
> This is what i am getting `Unknown (BOS+)` @UpstreamData 

That shouldn't be the case for that last function I sent, it should return a string with the model name and no suffix (`ANTMINER S19XP` is what it's supposed to be getting, but it's not or it would work properly).

**Comment by UpstreamData** [comment]
`await pyasic.miner_factory.get_miner_model_braiins_os(IP)`

**Comment by vijendtra** [comment]
Sorry my bad , you are correct it is returning `Antminer S19 XP`

**Comment by vijendtra** [comment]
Howver when i do this with same IP 

```
miner_ip= await get_miner(ip)

print(f"Miner Ip: {miner_ip}")print(f"Miner IP: {ip}, Firmware: {getattr(miner, 'firmware')}, Model:{getattr(miner, 'model')}",
```

it return me 

`Miner IP: 10.XX.XX.171, Firmware: BOS+, Model:Unknown (BOS+) Make: None`

**Comment by vijendtra** [comment]
@UpstreamData

**Comment by UpstreamData** [comment]
Fixed in 0.54.10.
